core/dbt/graph/selector_spec.py

```diff
@@ -116,14 +123,9 @@ def selection_criteria_from_dict(
         children_depth = _match_to_int(dct, 'children_depth')
 
         # If defined field in selector, override CLI flag
-        indirect_selection = dct.get('indirect_selection', None)
-        if indirect_selection:
-            if indirect_selection in ['eager', 'cautious']:
-                eagerly_expand = indirect_selection != 'cautious'
-            else:
-                raise RuntimeException(
-                    f'indirect_selection value "{indirect_selection}" is not valid!'
-                )
+        indirect_selection = dct.get('indirect_selection', None) or indirect_selection
+        if indirect_selection and indirect_selection not in ['eager', 'cautious']:
+            raise RuntimeException(f'indirect_selection value "{indirect_selection}" is invalid!')
```

Comment:
Rather than redefining the acceptable values here, we can use the same enum class for validation:
```python
        indirect_selection = IndirectSelection(
            dct.get('indirect_selection', None) or indirect_selection
        )
```
Then, the error raised is:
```
17:17:49 | [ error ] | Encountered an error:
'bad_value' is not a valid IndirectSelection
```
Instead of:
```
17:14:56 | [ error ] | Encountered an error:
Runtime Error
  Could not read selector file data: Runtime Error
    indirect_selection value "bad_value" is invalid!
```
That feels clear enough to me. If you wanted to keep the extra bit of context, we could try catching + wrapping the exception.

Reply:
Oh wow. Gonna be honest, I just didn't know the semantics around enums with confidence. Did some local testing and I find it a mental stretch that a StrEnum constructor _can take_ a StrEnum parameter. Not what I was expecting. Suffice to say, I'm onboard with this.